pom.xml

```diff
@@ -172,6 +172,33 @@
                     </includes>
                 </configuration>
             </plugin>
+            <!-- Shade 3rdparty libs to avoid classpath conflicts in unit tests -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <configuration>
+                    <artifactSet>
+                        <includes>
+                            <include>com.google.guava:*</include>
+                        </includes>
+                    </artifactSet>
+                    <createSourcesJar>true</createSourcesJar>
+                    <relocations>
+                        <relocation>
+                            <pattern>com.google.common</pattern>
+                            <shadedPattern>sling-xss.com.google.common</shadedPattern>
```

Comment:
Is this shadedPattern valid? Java package names cannot contain the `-` / dash character.

( https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-Identifier , https://docs.oracle.com/javase/specs/jls/se16/html/jls-7.html#jls-7.4.1 )

Reply:
i stumbled over this as well when taking over the configuration from sling models impl bundle - it seems to work fine there, and i deployed this bundle in a local instance and it seems to work there as well. but we can also easily remove the dash here.

Reply:
removed the dash

Reply:
Thanks for the change. I think this is a restriction of Java at the source level, but it seems to be legal bytecode. Before your change, the compiled java classes reference the 'sling-xss' package, see below snippet

```
$ javap -verbose -cp target/org.apache.sling.xss-2.3.9-SNAPSHOT.jar org.apache.sling.xss.impl.HtmlSanitizer
(...)
        47: invokespecial #86                 // Method org/owasp/html/DynamicAttributesSanitizerPolicy."<init>":(Lorg/owasp/html/HtmlStreamEventReceiver;Lsling-xss/com/google/common/collect/ImmutableMap;Lsling-xss/com/google/common/collect/ImmutableSet;Ljava/util/Map;Ljava/util/List;)V
(...)
````

More a curiosity than something to spend more time about.